audio.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+import simpleaudio as sa
+import scales
+from typing import Dict, List
+import keyword
+
+ALL_NOTES = scales.ALL_NOTES
+EVENT_DURATION = 0.25  # [seconds]: the time each note is played.
+SAMPLE_RATE = 44100
+LOWEST_A = 220  # [Hz]: frequency of A3, which is chosen to be the reference frequency, meaning it is the lowest note to be played.
+
+# Calculate frequencies and store them in two dicts for two intervals. This is
+# needed to play scales that smoothly increases or decreases.
+
+FREQUENCY_DICT = {}
+FREQUENCY_DICT_NEXT_OCTAVE = {}
+for note in ALL_NOTES:
+    frequency = LOWEST_A * 2 ** (ALL_NOTES[note] / 12)
+    FREQUENCY_DICT[note] = frequency
+    FREQUENCY_DICT_NEXT_OCTAVE[note] = 2 * frequency
+
+
+def slice_index(scale: List[str]) -> int:
+    slice_index = 7  # Ensures that the 8th note is an octave higher than the root note.
+    previous_note_index = 0
+    for note in scale[:-1]:  # Checks for slice_index prior to the 8th note.
+        note_index = ALL_NOTES[note]
+        if note_index < previous_note_index:
+            slice_index = scale.index(note)
+        previous_note_index = note_index
+    return slice_index
+
+
+def frequencies(scale: List[str], slice_index) -> List[float]:
+    """Translates a scale to a list of frequencies."""
+    notes_1st_octaves = scale[:slice_index]
+    notes_2st_octaves = scale[slice_index:]
```

Comment:
`notes_2nd_octave` probably?

Reply:
These variables (and the need to slice the scale) are removed in the latest refactor.